### ISSUE TITLE
Layers don't show up after 2nd page load

### ISSUE THREAD

**Issue by gergely-xyz** [AlexLavoie42/Nuxt-Mapbox#9]
If I define a source and layers in the template the layers don't show up on the map if I navigate away from the page with the map and come back. Refreshing the page shows the layer again, so does refreshing on an other page and coming back to the map. Using a function that triggers when the map loads works as expected.

This only shows the layer on first map load after refresh:
```vue
<template>
  <MapboxMap
    map-id="flights-map"
    class="h-100 w-100"
    :options="{
      style: mapStyle,
      center: [0, 0],
      zoom: 2,
    }"
  >

    <MapboxSource
      source-id="flights"
      :source="{
        type: 'geojson',
        data: flights_geojson,
      }"
    />

    <MapboxLayer
      source-id="flights"
      :layer="{
        id: 'unclustered-point',
        type: 'circle',
        source: 'flights',
      }"
    />

  </MapboxMap>
</template>
```

This works every time (with `@load="onMapLoad"` on the MapboxMap):
```vue
<script setup>
function onMapLoad(map) {
    map.addSource('flights-data', {
      type: 'geojson',
      data: flights_geojson,
    })

    map.addLayer({
      id: 'unclustered-point',
      type: 'circle',
      source: 'flights-data'
    })
  }
</script>
```

**Comment by gergely-xyz** [comment]
I've noticed the same for navigation and full screen control. 

```vue
<template>
  <MapboxMap
    map-id="flights-map"
    class="h-100 w-100"
    :options="{
      style: mapStyle,
      center: [0, 0],
      zoom: 2,
    }">
    <MapboxFullscreenControl :options="{}" />
    <MapboxNavigationControl :options="{}" />
  </MapboxMap>
</template>
```

**Comment by AlexLavoie42** [comment]
I wonder if this is an issue with the useMapbox composable. If you do this instead of using `@load`, do you get the same issue?

```js
useMapbox("flights-map", (map) => {
    map.addSource('flights-data', {
      type: 'geojson',
      data: flights_geojson,
    })

    map.addLayer({
      id: 'unclustered-point',
      type: 'circle',
      source: 'flights-data'
    })
});
```

**Comment by AlexLavoie42** [comment]
I am not able to reproduce this in stackblitz. Can you fork [the playground](https://stackblitz.com/github/AlexLavoie42/Nuxt-Mapbox?file=playground%2Fpages%2Findex.vue) and create a minimum reproduction? This would help me a lot in debugging the issue.

**Comment by gergely-xyz** [comment]
Check this out: https://stackblitz.com/edit/github-z7zukx?file=playground/pages/index.vue 
It's a simple two page nuxt app. If you use the NuxtLink on top to go to the "other" page and go back to the index the "MapboxFullscreenControl" and "MapboxNavigationControl" components in the top right corner of the map are no longer visible. The same happens with layers.

My initial idea for a workaround was to use the composable, but I got an error about the style not yet being loaded when I tried to add layers.

**Comment by AlexLavoie42** [comment]
Thanks for the help! After some investigation it looks like the issue comes from the existing map instance remaining on the client even though it was unloaded when changing page.

#11 should fix this bug, and I've published the fix on the dev branch (`1.2.3-dev.0`) if you would like to test that it works. I will publish the patch shortly after some more testing.

**Comment by gergely-xyz** [comment]
Awesome! Thank you for the quick response and the for the great module!
